### ISSUE TITLE
Fase 3: Entrenamiento y Evaluación

### ISSUE THREAD

**Issue by Serdan1** [Serdan1/cnn-cifar10#3]
Descripción:
En esta fase se entrenará la red neuronal convolucional (CNN) creada previamente con el dataset CIFAR-10, y se evaluará su rendimiento en el conjunto de prueba.
El objetivo es comprobar cómo el modelo aprende a reconocer patrones visuales a lo largo de las épocas y analizar su comportamiento mediante métricas de precisión y pérdida.

🎯 Objetivos

Compilar el modelo con los parámetros adecuados para clasificación multiclase.

Entrenar la CNN y registrar la evolución del aprendizaje.

Visualizar las curvas de entrenamiento y validación (precisión y pérdida).

Evaluar la precisión final sobre los datos de prueba.

Analizar el rendimiento general del modelo.

🧠 Tareas

 Compilar el modelo usando:

model.compile(optimizer='adam',
              loss='categorical_crossentropy',
              metrics=['accuracy'])

 Entrenar el modelo con .fit():

 Definir epochs=10 (mínimo recomendado).

 Usar validation_split=0.1 para validación automática.

 Guardar el historial del entrenamiento (history).

 Graficar la evolución del aprendizaje:

 Precisión de entrenamiento vs. validación.

 Pérdida de entrenamiento vs. validación.

 Incluir títulos, leyendas y etiquetas claras.

 Evaluar el modelo con .evaluate(x_test, y_test) y mostrar la precisión final.

 Analizar si existe sobreajuste (comparando entrenamiento y validación).

 Documentar todos los resultados y conclusiones en el Notebook.

📊 Ejemplo de código para gráficas
import matplotlib.pyplot as plt

plt.plot(history.history['accuracy'], label='Entrenamiento')
plt.plot(history.history['val_accuracy'], label='Validación')
plt.title('Precisión del modelo')
plt.xlabel('Épocas')
plt.ylabel('Precisión')
plt.legend()
plt.show()

plt.plot(history.history['loss'], label='Entrenamiento')
plt.plot(history.history['val_loss'], label='Validación')
plt.title('Pérdida del modelo')
plt.xlabel('Épocas')
plt.ylabel('Pérdida')
plt.legend()
plt.show()

✅ Criterios de finalización

El modelo se compila y entrena correctamente.

Las gráficas de precisión y pérdida se generan y se interpretan en Markdown.

Se reporta la precisión final en el conjunto de prueba (test_acc).

El análisis sobre el rendimiento y posible sobreajuste está documentado.
